### PR TITLE
Low code workflow naming standard

### LINKED ISSUE #338: Low Code Workflow Naming Standard
Our engineering standards are the things that we expect engineers to do. Look at the ['writing a standard' standard](https://engineering.homeoffice.gov.uk/standards/writing-a-standard/) and provide some brief information below

**What is the standard you are suggesting?**
Naming conventions and standards for Low Code Workflows, such as Azure Logic Apps, Power Automate and AWS Step Functions.

**What would be the benefit to the Home Office of adopting this standard?**
Standardisation of how to name variety of components within low-code workflow technologies. Primarily benefits maintainability by making things easier to read/understand.

**How might people follow it**
Ensure components such as workflow names, actions/steps, scope blocks etc are all named consistently and appropriately.

**Additional information**

**Please confirm the below**

- [x] I have looked at the writing-a-standard standard and think this would meet the standard
- [x] I have looked through our existing standards and think this is not covered elsewhere
- [x] I have checked through the open issues on the repository and this standard has not already been suggested

### PR DESCRIPTION
**Pull request by ARobinson26**
Is this pull request a content or a code change? (Please fill in the relevant section and delete the other)

# Code change
I can confirm:
## Accessibility considerations
- [ ] Please review the [accessibility checks for layout changes](https://github.com/UKHomeOffice/engineering-guidance-and-standards/blob/main/technical-docs/accessibility/layout-checks.md).

- [x] This change will not change layouts, page structures or anything else that might impact accessibility

or
- [ ] This change might impact accessibility, automated aXe tests cover the impact

or
- [ ] This change might impact accessibility and is not covered by automated aXe tests - manual testing has been performed
(If the change might impact accessibility then please add some further information here)

# Content change 
- [ ] Please review the [accessibility checks for content changes](https://github.com/UKHomeOffice/engineering-guidance-and-standards/blob/main/technical-docs/accessibility/content-checks.md).

I can confirm:
- [x] Content does not include any code or configuration changes (excluding frontmatter information)
- [x] Content meets the content standards
e.g. [Writing a principle](https://engineering.homeoffice.gov.uk/standards/writing-a-principle/) and [Writing a standard](https://engineering.homeoffice.gov.uk/standards/writing-a-standard/)
- [x] Content is suitable to open source, i.e.:
    - Content does not relate to unreleased gov policy
    - Content does not refer to anti-fraud mechanisms
    - Content does not include sensitive business logic
- [x] Last updated date for content is correct

### CODE REVIEW

**Comment by ARobinson26** [comment]
I'm unsure on whether we want to tag individual tools on standards. I feel like it would be a good idea to help find standards specific to the stack you're working with but appreciate it'd be difficult to keep maintained. Open to thoughts!

File: docs/standards/low-code-workflow-naming.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+---
+layout: standard
+order: 1
+title: Low code workflow naming
+date: 2023-11-01 # this should be the date that the content was most recently amended or formally reviewed
+id: SEGAS-00012 # Set unique ID for standard
```
Comment (edhamiltonHO) [inline docs/standards/low-code-workflow-naming.md:6]: We need to find a better way to assign SEGAS IDs to standards, we have a few PRs drafting but should probably only assign the ID once the standard is accepted. For now if you just want to mark this as SEGAS-TBC or similar

File: docs/standards/low-code-workflow-naming.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+## Requirement(s)
+
+<!-- Populate list for each requirement (there can be more than 2)-->
+
+- [Naming conventions MUST be consistent](#requirement-1)
+- [Acronyms or uncommon abbreviations MUST be defined in a central or project glossary](#requirement-2)
+- [Workflow, action step and scope block names MUST be descriptive and convey the purpose or objective](#requirement-3)
+- [Workflow, action step and scope block names MUST begin with a verb to describe the function, followed by the subject of the verb](#requirement-4)
+- [Words in a workflow name MUST be separated by a space where possible](#requirement-5)
+- [Scope blocks that are used for error handling MUST be prefixed with the word 'Try' or 'Catch'](#requirement-6)
```
Comment (edhamiltonHO) [inline docs/standards/low-code-workflow-naming.md:48]: Will need to update the links from `#requirement-n` to match the h3 text (but might not be worth doing this yet, if you are still confirming the requirement headings)

File: docs/standards/low-code-workflow-naming.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+### Naming conventions MUST be consistent
+
+Naming conventions must be applied consistently across a service. Consistency helps ensure maintainability and avoids confusion where multiple conventions are used in the same environment.
+
+### Acronyms or uncommon abbreviations MUST be defined in a central or project glossary
+
+Acronyms or abbreviations should only be used where it's not feasible or practical to use the full expansion. They should be defined in a either a central or project glossary accessible to all engineers on the service. Names with unknown acronyms or abbreviations can be meaningless and cause confusion for new engineers working on the service.
```
Comment (edhamiltonHO) [inline docs/standards/low-code-workflow-naming.md:56]: This is great, and should probably be applied more broadly beyond this standard!

**Comment by edhamiltonHO** [comment]
> I'm unsure on whether we want to tag individual tools on standards. I feel like it would be a good idea to help find standards specific to the stack you're working with but appreciate it'd be difficult to keep maintained. Open to thoughts!

Good question this, would be good to solidify a position on it. Some thoughts:

They are only tags, don't think we want to be too prescriptive on how they are used as long as they work and are useful. We should try to avoid hanging much site functionality off them

Tags should provide links between different content articles that relate to the same topic, 1 to many. Therefore I wouldn't expect us to create tags for things that would likely only relate to one content article (not that that is necessarily the case here, but might be).

The site search will pick up on all text in in a page, you don't need to create tags for that to work. You could instead refer to the tools in a list in the content body itself, this could lead to duplication though - e.g. you wouldn't want to be listing tools/stack in every low code standard.

**Comment by jeff-horton-ho-sas** [comment]
> The site search will pick up on all text in in a page, you don't need to create tags for that to work. You could instead refer to the tools in a list in the content body itself, this could lead to duplication though - e.g. you wouldn't want to be listing tools/stack in every low code standard.

If we found content was being added just to optimise search, we could add support for a searchTerms list in the page front matter and append it to the page content when generating the search index https://github.com/UKHomeOffice/engineering-guidance-and-standards/blob/main/_includes/layouts/search-index.njk#L9

**Comment by edhamiltonHO** [comment]
> > The site search will pick up on all text in in a page, you don't need to create tags for that to work. You could instead refer to the tools in a list in the content body itself, this could lead to duplication though - e.g. you wouldn't want to be listing tools/stack in every low code standard.
> 
> If we found content was being added just to optimise search, we could add support for a searchTerms list in the page front matter and append it to the page content when generating the search index https://github.com/UKHomeOffice/engineering-guidance-and-standards/blob/main/_includes/layouts/search-index.njk#L9

Yes, we did this with the standard `id` field in frontmatter, so people could search for content by that, e.g. `SEGAS-00001` or `00001`

**Comment by ARobinson26** [comment]
Addressed comments above from @edhamiltonHO. I've also removed the tech stacks from the tags for now. 👍🏼

**Review by edhamiltonHO — APPROVED** [review]

**Review by robertdeniszczyc2 — APPROVED** [review]

**Comment by edhamiltonHO** [comment]
Rebased in order to verify commits and nudge checks to complete

**Pull request merged**